File: scala-demo/src/main/scala/com/atguigu/chapter12/work3/Names.scala
package com.atguigu.chapter12.work3

/**
 * @author tianmin
 * @date 2020/2/22 0022
 * @notes
 */
class Names {

}

object Names{
  def unapplySeq(str: String): Option[Seq[String]] = {
    if(str.contains(",")){
      println("调用了")
      Some(str.split(","))
    }else{
      None
    }
  }
}

File: atguiguprojectct/atguigu-project-ct/ct-common/src/main/java/com/atguigu/ct/common/constant/Names.java
package com.atguigu.ct.common.constant;

import com.atguigu.ct.common.bean.Val;

/**
 * @author tianmin
 * @date 2020/2/1 0001
 * @notes
 */
public enum Names implements Val {
    NAMESPACE("ct"),
    TOPIC("ct"),
    TABLE("ct:calllog"),
    CF_CALLER("caller"),
    CF_CALLEE("callee"),
    CF_INFO("info"),
    JEDIS_USER("ct_user"),
    JEDIS_DATE("ct_date");

    private String name;
    private Names(String name){
        this.name = name;
    }


    public void setValue(Object val) {

    }

    public String getValue() {
        return name;
    }
}
